pkg/controllers/nodeclass/validation.go

```diff
@@ -191,74 +201,37 @@ func (v *Validation) validateRunInstancesAuthorization(
 	nodeClaim *karpv1.NodeClaim,
 	tags map[string]string,
 ) (reason string, requeue bool, err error) {
-	// NOTE: Since we've already validated the status conditions are true, these should never occur
-	if len(nodeClass.Status.AMIs) == 0 {
-		return "", false, fmt.Errorf("no resolved amis in status")
-	}
-	if len(nodeClass.Status.Subnets) == 0 {
-		return "", false, fmt.Errorf("no resolved subnets in status")
-	}
-	if len(nodeClass.Status.SecurityGroups) == 0 {
-		return "", false, fmt.Errorf("no resolved security groups in status")
-	}
-	if nodeClass.Status.InstanceProfile == "" {
-		return "", false, fmt.Errorf("no instance profile in status")
+	opts, err := v.mockOptions(ctx, nodeClaim, nodeClass, tags)
+	if err != nil {
+		return "", false, fmt.Errorf("generating options, %w", err)
 	}
 
-	var instanceType ec2types.InstanceType
-	requirements := scheduling.NewNodeSelectorRequirements(nodeClass.Status.AMIs[0].Requirements...)
-
-	instanceType = ec2types.InstanceTypeM6gLarge
-	if requirements.Get(corev1.LabelArchStable).Has(karpv1.ArchitectureAmd64) {
-		instanceType = ec2types.InstanceTypeM5Large
+	// We can directly marshal from CreateLaunchTemplate LaunchTemplate data
+	runInstancesInput := &ec2.RunInstancesInput{}
+	raw, err := json.Marshal(launchtemplate.GetCreateLaunchTemplateInput(ctx, opts[0], corev1.IPv4Protocol, "").LaunchTemplateData)
+	if err != nil {
+		return "", false, fmt.Errorf("converting launch template input to run instances input, %w", err)
+	}
+	if err = json.Unmarshal(raw, runInstancesInput); err != nil {
+		return "", false, fmt.Errorf("converting launch template input to run instances input, %w", err)
```

Comment:
This is to make struct assignment for `RunInstancesInput` easy, right?

Reply:
Yep, should copy over everything, aside from the overrides that we specify with fleet -- those we manually need to set here